src/navigator/NavigationService.ts

```diff
@@ -45,6 +47,23 @@ async function ensureNavigator() {
   }
 }
 
+export const popToScreen: SafeNavigate = (...args) => {
+  const [routeName] = args
+  ensureNavigator()
+    .then(() => {
+      Logger.debug(`${TAG}@popToScreen`, `Dispatch ${routeName}`)
+      while (
+        navigationRef.current?.canGoBack() &&
+        navigationRef.current?.getCurrentRoute()?.name !== routeName
+      ) {
+        navigationRef.current?.dispatch(StackActions.pop())
+      }
+    })
+    .catch((reason) => {
+      Logger.error(`${TAG}@popToScreen`, 'Navigation failure', reason)
```

Comment:
`Logger.warn`? do we need to alert Sentry here?

Reply:
The other navigation functions use logger.error when failing to navigate, so I think we do want to alert sentry here to be consistent.